src/main.rs

```diff
@@ -324,7 +324,15 @@ fn create_app<'a, 'b>(
         .arg(
             Arg::with_name("api-socket")
                 .long("api-socket")
-                .help("HTTP API socket path (UNIX domain socket).")
+                .help("HTTP API socket (UNIX domain socket).")
+                .takes_value(true)
+                .min_values(1)
+                .group("vmm-config"),
+        )
+        .arg(
+            Arg::with_name("api-socket-fd")
```

Comment:
I'd be in favor of breaking compatibility and extend `api-socket` instead. I would introduce `path=` for the current way of passing a socket path, and `fd=` for the new way of providing the file descriptor directly.
@rbradford wdyt?

Reply:
FYI my attempt at the single --api-socket argument is https://github.com/cloud-hypervisor/cloud-hypervisor/commit/be673701a0fdcb69ae673655f29db0d7640020bf

Reply:
We would need to do a transition. e.g. if it looks likes new style use it like that else fallback to using it as a path for at least a couple of releases.

Reply:
Okay, I'll open a new PR based on that idea while we see what happens with the micro-http PR.